### PR TITLE
Raising NotImplementedError for elements that actually are not implemented.

### PR DESCRIPTION
**Pull request by itholic**
we're raising `ValueError` for some 'NotImplemented' elements.

i think it's better to raise `NotImplementedError` rather than `ValueError` for those elements so that we can find more easily when we want to implement them.

### CODE REVIEW

**Comment by itholic** [comment]
while i'm here, i find more cases.

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -5458,9 +5458,10 @@ def sort_index(self, axis: int = 0,
         b 1  0  3
         """
         if axis != 0:
-            raise ValueError("No other axes than 0 are supported at the moment")
+            raise NotImplementedError("No other axis than 0 are supported at the moment")
```
Comment (ueshin) [inline databricks/koalas/frame.py]: Should we use `validate_axis` and support `'index'` as well while we are here?

Comment (itholic) [reply databricks/koalas/frame.py]: yeah, we'd better do that. i'll check if another place to add, too.

thanks for the comment! :)

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**